src/main/java/fi/evident/elasticsearch/voikko/analysis/VoikkoTokenFilter.java

```diff
@@ -128,4 +139,82 @@ private void outputAlternative(String token) {
+        // Contains code from the Voikko Filter for Solr
+        // by the National Library of Finland.
+        //
+        // https://github.com/NatLibFi/SolrPlugins
+        Set<String> compoundForms = new LinkedHashSet<String>();
+
+        for (Analysis analysis: analysisList) {
+            if (!analysis.containsKey("WORDBASES")) {
+                continue;
+            }
+            String wordbases = analysis.get("WORDBASES");
+            // Split by plus sign (unless right after an open parenthesis)
+            String matches[] = wordbases.split("(?<!\\()\\+");
+
+            int currentPos = 0, lastPos = 0;
+            String lastWordBody = "";
+            assert matches.length > 1;
+            // The string starts with a plus sign, so skip the first (empty) entry.
+            for (int i = 1; i <= matches.length - 1; i++) {
+                String wordAnalysis, wordBody, baseForm;
+
+                // Get rid of equals sign in e.g. di=oksidi.
+                wordAnalysis = matches[i].replaceAll("=", "");;
+                int parenPos = wordAnalysis.indexOf('(');
+                if (parenPos == -1) {
+                    wordBody = baseForm = wordAnalysis;
+                } else {
+                    // Word body is before the parenthesis
+                    wordBody = wordAnalysis.substring(0, parenPos);
+                    // Base form or derivative is in parenthesis
+                    baseForm = wordAnalysis.substring(parenPos + 1, wordAnalysis.length() - 1);
+                }
+
+                String word;
+                int wordOffset, wordLen;
+                boolean isDerivative = baseForm.startsWith("+");
+                if (isDerivative) {
+                    // Derivative suffix, merge with word body
+                    word = lastWordBody + wordBody;
+                    wordOffset = lastPos;
+                    wordLen = word.length();
+                } else {
+                    word = baseForm;
+                    wordOffset = currentPos;
+                    wordLen = word.length();
+                    lastWordBody = wordBody;
+                    lastPos = currentPos;
+                    currentPos += baseForm.length();
+                }
+
+                // Make sure we don't exceed the length of the original term
+                int termLen = charTermAttribute.toString().length();
+                if (wordOffset + wordLen > termLen) {
+                    if (wordOffset >= termLen) {
+                        wordOffset = wordLen - termLen;
+                        if (wordOffset < 0) {
+                            wordOffset = 0;
+                        }
+                    } else {
+                        wordLen = termLen - wordOffset;
+                    }
+                }
+
+                int maxSubwordSize = cfg.maximumSubwordSize;
+                int minSubwordSize = cfg.minimumSubwordSize;
+                if (wordLen > minSubwordSize) {
```

Comment:
If I specify minimum size to be 2, I'd expect that words of length 2 will be included, but this excludes them. So perhaps this should be `>=` instead of `>`?